### ISSUE TITLE
How to test the trained model in new tasks?

### ISSUE THREAD

**Issue by morantumi** [jaesik817/bmaml_rl#4]
I have a question is that if you do test with the trained model. If the answer is positive, could you share the method. I train BMAML in Ant for 200 times and draw the curve with the max average return of 5 particles. But how can I test the model in new tasks?  I try to make the "load_policy" is true but the result is not right. So could you tell me how to deal with it?

**Comment by morantumi** [comment]
I think the function of "load_policy" is to comtinue or extand the tranin times. For example, if I have trained 30 times and want to add 20 times, I can load the itr_30.pkl of the 30 times' and set up the meta-itr=50, is that right?
But how to test the trained model? I am really confused with it. Hope for your reply. Thank you very much.

**Comment by morantumi** [comment]
I trained the model in Ant with the command
`python bmaml_examples/locomotion/bmaml_ant.py --num_particles=5 --num_parallel=10 --fast_lr=1.0 --meta_step_size=0.01 --fast_batch_size=50 --meta_batch_size=20 --meta_iter=250 --meta_method=chaser --method=svpg --svpg_alpha=0.1`
The random seed is 1 and the result is as follow. As we can see, the result is not very better, I wonder if there is some more reasonable parameters. I don't know the reason why the model will get worse when they in about 150 itr. 
![chaser_vs_trpo_antvel-200_fbs50_mbs20_flr_1 0_mlr_0 01_summary2_seaborn-paper](https://user-images.githubusercontent.com/46220439/68441865-6ecbcd00-020a-11ea-9d8c-a9e4086154ea.png)

Any ideas on how to fix the issue or work around it?
Thank you.

**Comment by morantumi** [comment]
I change  the `seed=0` and now have the result as follow. The result is similar. In my opinion,  I think maybe the `fast_lr=1.0` is too high which result to overfiting. I really have no idea why the result is so bad. The model trained by maml is better than Bmaml's and the tendency of maml is right..... 
![chaser_vs_trpo_antvel-200_fbs50_mbs20_flr_1 0_mlr_0 01_summary1_seaborn-paper](https://user-images.githubusercontent.com/46220439/68444037-a63d7800-0210-11ea-91a9-434e518020b1.png)

**Comment by jsikyoon** [comment]
Hi Morantumi,

Thank you for your interesting about our project.

Let me explain from the beginning.

>>how to test the trained model?

we didn't implement the test code separately. As you mentioned, the 'load_policy' configuration is to continue training from the finished one before. But it is not working correctly when I checked so I commented 'it is not working.'

 >>Ant experiment

Yes, I also think fast learning rate is too big when comparing my history. And, I'm not sure which option you choose (task option), if you select goal direction, on the goal direction task, our model showed high fluctuation as our paper (but it outperformed maml). 

If you can share the details, then let me test on my side too.

Thanks.
Regards, Jaesik.

**Comment by morantumi** [comment]
Thank you very much for your reply!
First, I choose the task：
> ant_env_rand_goal
The reason why I choose the `fast_lr=1.0` is that I get good result when I use the method of MAML with this value in environmnet `ant_env_rand_goal`. The parameters setting in MAML is  that
`fast_lr=1.0  meta_lr=0.01 kl_weighting=0.1 `

So I implement the BMAML with the same `fast_lr` and `meta_lr`. Maybe it causes the problem. 

Besides, I do not clear the impact of the parameter `SVPG_lr`,  could you tell me the specific value of this parameter when trained in the locomotion environment. The results of BMAML I showed  was obtained in `SVPG_lr=0.1.` I notice that the computation code about the SVPG is as follow, so I think it may also have a major impact on the final result. If I should change it to 1.0? Could you tell me the vaule of this parameters.
`grad=(tf.matmul(kernel_mat,((1/self.svpg_alpha)*f_g_list))-grad_kernel)/len(self.policy_list)`

PS:
The result of MAML in  ant_env_rand_goal is as follow:
<img width="511" alt="屏幕快照 2019-11-08 11 46 32" src="https://user-images.githubusercontent.com/46220439/68448253-db03fc00-021d-11ea-9e3f-a095463dc2ba.png">

**Comment by jsikyoon** [comment]
Could I know the running script for MAML?

**Comment by morantumi** [comment]
Of course.
I use the code from [https://github.com/cbfinn/maml_rl](url), also is implemented in Rllab.

The aglo is `sandbox.rocky.tf.algos.maml_trpo`
The policy is `maml_minimal_gauss_mlp_policy`
The environment is `ant_env_rand_goal`
The script is `python train.py Maml --env Ant --fast_batch_size=50 --meta_batch_size=20 --fast_learning_rate=1 --meta_learning_rate=0.01 --seed=1 --kl_weighting=0.1  init_std=1`
I also set  `num_total_tasks=100  num_grad_updates=1 n_itr=300`

**Comment by jsikyoon** [comment]
The maml uses TRPO, which is powerful and expensive algorithm for reinforcement learning. As we described in our paper, we thus compared emaml-trpo vs bmaml-trpo and emaml-reptile vs bmaml-chaser.

**Comment by morantumi** [comment]
Thanks very much for your reply.
It's my fault to ignore this. I will change the maml-trpo to the maml-reptile and then do the experiment again.  I do this comparement becuase the following result in the paper. I wil retry the experiment.
<img width="811" alt="屏幕快照 2019-11-08 13 42 35" src="https://user-images.githubusercontent.com/46220439/68452605-00006b00-022e-11ea-9346-774bdd48597b.png">
Besides, I confused with the value setting of `fast_lr` and `svpg_lr` for the `ant_env_rand_goal` for a long time. I has tried  `fast_lr=1.0` and `svpg_lr=0.1` but the result is bad.  It takes many days to complete 200 times itr. 
So, could I know the specific values for these two parameters for Ant tasks?
Hope for your advice and thank you very much.

**Comment by jsikyoon** [comment]
Yon can check that in running_scripts.txt

**Comment by morantumi** [comment]
Thank you for your reply! It is really very helpful. Thank you very much.
